### PR TITLE
Use indexing instead of reshape for broadcasting

### PR DESCRIPTION
**Pull request by ngimel**
This is needed for MLIR rewrite
This replaces
```
xindex = xoffset + tl.reshape(tl.arange(0, XBLOCK), [XBLOCK, 1])
``` 
with
```
xindex = xoffset + tl.arange(0, XBLOCK)[:, None]
```
so code is a bit more readable, and compiles with master triton (which doesn't currently support first construct). 

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -933,17 +933,17 @@ def reduction(self, name, dtype, src_dtype, reduction_type, index, value):
...
                 self.suffix.writelines(
                     [
-                        f"{accumulator_index}_reduce = tl.reshape(",
-                        f"\ttl.{reduction_type}({accumulator}, {dim}), [{', '.join(sizes)}]).to(tl.int32)",
-                        f"{accumulator_index}_mask = (tl.reshape(tl.arange(0, {reduction_range_prefix.upper()}BLOCK),",
-                        f"\t[{', '.join(reduction_sizes)}]) == {accumulator_index}_reduce)",
-                        f"{result_var} = tl.reshape(tl.sum(",
-                        f"\ttl.where({accumulator_index}_mask, {accumulator_index}, 0), {dim}), [{', '.join(sizes)}])",
+                        f"{accumulator_index}_reduce = "
+                        f"tl.{reduction_type}({accumulator}, {dim})[{', '.join(sizes)}].to(tl.int32)",
```
Comment (desertfire) [inline torch/_inductor/codegen/triton.py:937]: Should we keep `\t` for the readability of the generated code?

**Review by desertfire — APPROVED** [review]

File: torch/_inductor/codegen/triton.py (hunk 1 above)
Comment (ngimel) [reply torch/_inductor/codegen/triton.py:937]: The generated code now puts things on a single line (not absence of `,` at the end of f-string lines, which, IMO, is more readable than previously:
```
   _tmp1_index_reduce = tl.argmax(_tmp1, 1)[:, None].to(tl.int32)
    _tmp1_index_mask = tl.arange(0, RBLOCK)[None, :] == _tmp1_index_reduce
    tmp1 = tl.sum(tl.where(_tmp1_index_mask, _tmp1_index, 0), 1)[:, None]
```
compared to 
```
   _tmp1_index_reduce = tl.reshape(tl.argmax(_tmp1, 1), [XBLOCK, 1]).to(tl.int32)
    _tmp1_index_mask = (tl.reshape(tl.arange(0, RBLOCK),
        [1, RBLOCK]) == _tmp1_index_reduce)
    tmp1 = tl.reshape(tl.sum(
        tl.where(_tmp1_index_mask, _tmp1_index, 0), 1), [XBLOCK, 1])
 ```

**Pull request closed without merging**